README.md

```diff
@@ -1,16 +1,235 @@
+`,` again.
+
+[CSV] doesn't say which data has to come first or second: just that it's
+divided into rows by `\n` and the fields are separated by commas. We could have
+put the name first or the id looking thing last. Both would have been valid
+CSV. We'll explain the specifics of this file as we get into the lab work
+below.
+
+## Parse Data from a file
+
+We can use Ruby's built-in [CSV] library to interpret the `.csv` file according
+to the "comma-separated values" format.
+
+In this lab you'll create a class that manages a CSV file. It will turn "data
+from the disk" into "rows of CSV-foratted data." Furthermore, the class will
+provide analysis of the data it possesses (augmenting the powerless file like
+how a mecha armor augments a green-haired anime hero!).
+
+![Voltron is a wrapping class to humans](https://media.giphy.com/media/niNOWsn2lTxXq/giphy.gif)
+
+_The lion mechs are "wrappers" to humans in this analogy_
+
+## Create Object-Oriented Program
+
+### Scenario
+
+Let's suppose we're working at a certain magical high school of wizard-like
+studies. Our Department of Luck Studies (DoLS) is looking for a new
+President. To do this, they had candidates throw many dice rolls in hopes of
+finding the luckiest individual. _Unluckily_, the database they were storing
+this information in crashed and lost most of its records. Our dedicated IT
+team, assisted by Unix magic and the powerful spells `grep` and `awk`, however,
+managed to save 100 of the trials to the file `trials.csv`.
+
+The rows in the CSV represent:
+
+```csv
+1,Arya,6,"4,3"
+```
+
+1.  An `id` for the trial
+2.  A participant name (a candidate)
+3.  The maximum number of pips on the die thrown in the next field
+4.  The results of the candidate's throw. Multiple die are separated by a `,`
+
+So the line above says, in English, "In trial 1, Arya threw six sided dice.
+There were two of them (splitting the last field and counting the numbers),
+their pips were 4 and 3 leading to a total pip-value of `7`."
+
+### First Release: Learn About the Data
+
+We want to analyze this file for the DoLS.
+
+The tests will guide you in creating a class called `LuckAnalyzer.rb` in
+`lib/luck_analyzer.rb`.
+
+- `LuckAnalyzer` class will:
+  - Be initialized by being passed the name of the file to read. It must also
+    end in `.csv`. Else we `raise` an `ArgumentError`
+  - `initialize`
+    - Will build a file path from the file name passed in (starter code is
+      provided)
+    - Will "parse" the data in the file at the file path according to the CSV
+      format using the Ruby [CSV] library method that reads in a file and
+      converts the rows to an `Array` of `Arrays`. Consult the
+      [documentation][csv] for this method
+    - The parsed CSV data should be available through a reader called
+      `csv_data`
+  - Instance methods
+    - `#common_number_of_trials`: Because the database crashed, we're not sure
```

Comment:
I think this still needs some clarity. Maybe a narrative example - i.e. we've got 4 trials from Arya, 5 trials from Belinda, and 3 trials from Carl, [A, A, A, A, B,
      B, B, B, B, C, C, C]. Since Carl has the fewest,...

Reply:
Updated in ece41a6